src/app/app.component.ts

```diff
@@ -13,16 +14,22 @@ import { AuthActions } from './auth/action-types';
   styleUrl: './app.component.scss',
 })
 export class AppComponent implements OnInit {
-  title = 'B-commerce';
+  public title = 'B-commerce';
+  private authState = this.store.selectSignal(selectAuthState);
 
-  count$?: Observable<number>;
+  constructor(private store: Store<AppState>, private localStorageService: LocalStorageService) {
+  }
 
-  constructor(private store: Store<AppState>) {}
+  @HostListener('window:beforeunload', ['$event'])
+  public beforeUnload() {
+    this.localStorageService.setItem('auth', JSON.stringify(this.authState()));
+  }
 
   ngOnInit(): void {
-    const userProfile = localStorage?.getItem('user') ?? '';
+    const userProfile = this.localStorageService.getItem('user') as User;
     if (userProfile) {
-      this.store.dispatch(AuthActions.login({ user: JSON.parse(userProfile) }));
+      this.store.dispatch(AUTH_ACTIONS.getAuthState(userProfile));
```

Comment:
The name says `getAuthState`. Are you sure you are getting?

Reply:
The naming isn't a good naming convention,
I think something like updateAuthState would sound right